pkg/downloader/manager.go

```diff
@@ -78,6 +78,32 @@ type Manager struct {
 	RepositoryCache  string
 }
 
+func NewManager(
```

Comment:
This is new public method does not appear to do more than taking in fields and returning them wrapped in a Manager. I think this is unnecessary

Reply:
This exposes unwanted behavior caused by not specifying some of these arguments when creating this struct. If there is a better pattern for exposing errors like the ones that caused this bug I'm all for it.